flashinfer/gemm.py

```diff
@@ -1828,14 +1842,17 @@ def mm_fp4(
         Out tensor, shape (m, n), bf16 or fp16, defaults to ``None``.
 
     block_size: int
-        Block size for FP4 quantization, only 16 is supported.
+        Block size for FP4 quantization, only 16 and 32 are supported.
 
     use_8x4_sf_layout: bool
         Whether to use 8x4 scale factor layout or 128x4 scale factor layout, defaults to False.
 
     backend: Literal["cudnn", "trtllm", "cutlass"]
         Backend to use, defaults to "cudnn".
 
+    use_nvfp4: bool
```

Comment:
How about inferring from scale data types instead?
When descale tensor has data type `float8_e4m3fn`, uses mxfp4, when descale tensor has data type `float8_e8m0fnu` (currently represented by int8 instead because e8m0 is only available from torch 2.8 on).